# <center> **Kpis de Siniestros Viales**

Los KPIs, o Indicadores Clave de Proceso, son herramientas cruciales que evalúan la eficacia en la consecución de objetivos fundamentales. Estos indicadores representan valores que cuantifican el logro de metas, generalmente expresadas en índices o porcentajes. A diferencia de las métricas, que son mediciones de datos en sí mismas sin un objetivo específico, los KPIs se distinguen al medir metas clave específicas, no cualquier aspecto. Además, los KPIs requieren métricas para su cálculo, pero no todas las métricas son indicadores, y no todos los indicadores son KPIs.

## Importaciones

In [105]:
import numpy as np
import pandas as pd


## Extraccion de la informacion

In [106]:
data = pd.read_csv('Homicidios.csv')
data

Id                   ROL       SEXO  EDAD  # Victimas       Fecha  \
0    2016-0001             CONDUCTOR  MASCULINO    19           1  2016-01-01   
1    2016-0002             CONDUCTOR  MASCULINO    70           1  2016-01-02   
2    2016-0003             CONDUCTOR  MASCULINO    30           1  2016-01-03   
3    2016-0004             CONDUCTOR  MASCULINO    18           1  2016-01-10   
4    2016-0005             CONDUCTOR  MASCULINO    29           1  2016-01-21   
..         ...                   ...        ...   ...         ...         ...   
712  2021-0092                PEATON   FEMENINO    50           1  2021-12-12   
713  2021-0093  PASAJERO_ACOMPAÑANTE   FEMENINO    18           1  2021-12-13   
714  2021-0094  PASAJERO_ACOMPAÑANTE   FEMENINO    43           1  2021-12-20   
715  2021-0095             CONDUCTOR  MASCULINO    27           1  2021-12-30   
716  2021-0096             CONDUCTOR  MASCULINO    60           1  2021-12-15   

      Año  Mes  Dia      Hora  ...  Comuna  \
0    2016    1    1  04:00:00  ...       8   
1    2016    1    2  01:15:00  ...       9   
2    2016    1    3  07:00:00  ...       1   
3    2016    1   10  00:00:00  ...       8   
4    2016    1   21  05:20:00  ...       1   
..    ...  ...  ...       ...  ...     ...   
712  2021   12   12  06:20:00  ...       3   
713  2021   12   13  17:10:00  ...       7   
714  2021   12   20  01:10:00  ...       9   
715  2021   12   30  00:43:00  ...      11   
716  2021   12   15  10:30:00  ...       9   

                                   XY (CABA)         Pos x         Pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
712  Point (105258.35368554 102122.93231400)  -58.40596860  -34.61011987   
713   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
714    Point (99624.29795829 97569.69801131)  -58.46739825  -34.65117757   
715   Point (99116.45492358 101045.23284826)  -58.47293407  -34.61984745   
716    Point (99324.54463985 97676.26932409)  -58.47066794  -34.65021673   

      Participantes Victima    Acusado Semestre Día semana Nombre día  
0         MOTO-AUTO    MOTO       AUTO        1          4    Viernes  
1    AUTO-PASAJEROS    AUTO  PASAJEROS        1          5     Sábado  
2         MOTO-AUTO    MOTO       AUTO        1          6    Domingo  
3           MOTO-SD    MOTO         SD        1          6    Domingo  
4    MOTO-PASAJEROS    MOTO  PASAJEROS        1          3     Jueves  
..              ...     ...        ...      ...        ...        ...  
712     PEATON-AUTO  PEATON       AUTO        2          6    Domingo  
713       MOTO-AUTO    MOTO       AUTO        2          0      Lunes  
714       MOTO-AUTO    MOTO       AUTO        2          0      Lunes  
715     MOTO-CARGAS    MOTO     CARGAS        2          3     Jueves  
716     AUTO-CARGAS    AUTO     CARGAS        2          2  Miércoles  

[717 rows x 26 columns]

In [107]:
data.head()

Id        ROL       SEXO  EDAD  # Victimas       Fecha   Año  Mes  \
0  2016-0001  CONDUCTOR  MASCULINO    19           1  2016-01-01  2016    1   
1  2016-0002  CONDUCTOR  MASCULINO    70           1  2016-01-02  2016    1   
2  2016-0003  CONDUCTOR  MASCULINO    30           1  2016-01-03  2016    1   
3  2016-0004  CONDUCTOR  MASCULINO    18           1  2016-01-10  2016    1   
4  2016-0005  CONDUCTOR  MASCULINO    29           1  2016-01-21  2016    1   

   Dia      Hora  ...  Comuna                                XY (CABA)  \
0    1  04:00:00  ...       8    Point (98896.78238426 93532.43437792)   
1    2  01:15:00  ...       9    Point (95832.05571093 95505.41641999)   
2    3  07:00:00  ...       1   Point (106684.29090040 99706.57687843)   
3   10  00:00:00  ...       8    Point (99840.65224780 94269.16534422)   
4   21  05:20:00  ...       1  Point (106980.32827929 100752.16915795)   

          Pos x         Pos y   Participantes Victima    Acusado Semestre  \
0  -58.47533969  -34.68757022       MOTO-AUTO    MOTO       AUTO        1   
1  -58.50877521  -34.66977709  AUTO-PASAJEROS    AUTO  PASAJEROS        1   
2  -58.39040293  -34.63189362       MOTO-AUTO    MOTO       AUTO        1   
3  -58.46503904  -34.68092974         MOTO-SD    MOTO         SD        1   
4  -58.38718297  -34.62246630  MOTO-PASAJEROS    MOTO  PASAJEROS        1   

  Día semana Nombre día  
0          4    Viernes  
1          5     Sábado  
2          6    Domingo  
3          6    Domingo  
4          3     Jueves  

[5 rows x 26 columns]

# ***Primer KPI***

**Reducir en un 10% la tasa de homicidios en siniestros viales de los últimos seis meses, en CABA, en comparación con la tasa de homicidios en siniestros viales del semestre anterior**

Definimos a la tasa de homicidios en siniestros viales como el número de víctimas fatales en accidentes de tránsito por cada 100,000 habitantes en un área geográfica durante un período de tiempo específico. Su fórmula es: (Número de homicidios en siniestros viales / Población total) * 100,000

Se realiza Web Scraping para encontrar la poblacion de CABA,en los años requeridos.


Para el año 2016 se cuenta con una Poblacion de 3.059.122 (chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://www.argentina.gob.ar/sites/default/files/siempro_-_caba_10.07.17.pdf)

No se encontro informacion de los años 2017 y 2018, se procede a hacer un promedio de los años 2016 y 2019 para trabajar con estos datos, dando como resultado 3.065.575 habitantes.

Para el año 2019 se cuenta con una Poblacion de 3.072.029 (chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://www.argentina.gob.ar/sites/default/files/191202_caba.pdf)

Para el año 2020 habian 3.075.646 habitantes (https://www.argentina.gob.ar/sites/default/files/2021/09/reporte_caba_final.pdf)  

Se hace un promedio entre las dos poblaciones para encontrar la poblacion del 2021 y se obtiene una poblacion estimada de 3.098.676

Para el año 2022 un aproximado de 3.121.707 (https://www.argentina.gob.ar/pais/poblacion/centros). 
 

Creamos un diccionario con la poblacion por año

In [108]:
poblacion_por_año = { 2016: 3059122,
    2017: 3065575 ,
    2018: 3065575 ,
    2019: 3072029,
    2020: 3075646,
    2021: 3121707}

In [109]:
#Creamos la tabla con los datos requeridos en el kpi

kpi1= data.groupby(["Año", "Semestre"])["# Victimas"].sum().reset_index()

#Introducimos el diccionario de poblacion que creamos anteriormente
kpi1["Poblacion"] = kpi1["Año"].map(poblacion_por_año)

# Calculamos la tasa de homicidios
kpi1["Tasa"] = ((kpi1["# Victimas"] / kpi1["Poblacion"]) * 100000).round(2)

# Revisamos en términos porcentuales los cambios
kpi1["Cambio Porcentual"] = (kpi1["Tasa"].pct_change() * 100).round(2)


kpi1


Año  Semestre  # Victimas  Poblacion  Tasa  Cambio Porcentual
0   2016         1          67    3059122  2.19                NaN
1   2016         2          83    3059122  2.71              23.74
2   2017         1          81    3065575  2.64              -2.58
3   2017         2          79    3065575  2.58              -2.27
4   2018         1          76    3065575  2.48              -3.88
5   2018         2          85    3065575  2.77              11.69
6   2019         1          59    3072029  1.92             -30.69
7   2019         2          47    3072029  1.53             -20.31
8   2020         1          31    3075646  1.01             -33.99
9   2020         2          56    3075646  1.82              80.20
10  2021         1          55    3121707  1.76              -3.30
11  2021         2          42    3121707  1.35             -23.30

Revisamos en que año y semestre si se cumplio con el kpi

In [110]:
kpi1[kpi1["Cambio Porcentual"]<-10].reset_index()

index   Año  Semestre  # Victimas  Poblacion  Tasa  Cambio Porcentual
0      6  2019         1          59    3072029  1.92             -30.69
1      7  2019         2          47    3072029  1.53             -20.31
2      8  2020         1          31    3075646  1.01             -33.99
3     11  2021         2          42    3121707  1.35             -23.30

Aunque se observaron disminuciones en la tasa de homicidios en siniestros viales en algunos periodos, solo en ciertos casos se logró cumplir con el KPI de reducir en un 10% la tasa en comparación con el semestre anterior.

Año 2017:

Tanto en el primer como en el segundo semestre, se experimentaron reducciones, pero ninguna alcanzó el 10%.

Año 2018:

Similar al año 2017, hubo reducciones en ambos semestres, pero ninguna cumplió con el KPI del 10%.

Año 2019:

Ambos periodos mostraron disminuciones de más del 20%, superando significativamente el objetivo del KPI.

Año 2020:

En el primer semestre de 2020, se cumplió el KPI, pero en el segundo semestre, la tasa aumentó en un 80.65%, incumpliendo el objetivo.

Año 2021:

Aunque en el primer semestre se logró una disminución del 3.23%, en el segundo semestre la tasa disminuyó en un 23.64%, cumpliendo con el KPI.
En resumen, si bien algunos periodos mostraron mejoras en la seguridad vial, solo en los años 2019 y 2021 se lograron reducciones consistentes de más del 10%, cumpliendo así con el objetivo establecido por el KPI.

In [111]:
datakpi1 = kpi1.groupby(["Año", "Semestre"])[["# Victimas", "Poblacion", "Cambio Porcentual"]].sum().reset_index()
datakpi1

datakpi1.to_csv('datakpi1.csv',index=False, encoding='utf-8')
print(f'Se guardó el archivo ')

Se guardó el archivo 


#  ***Segundo KPI***

**Reducir en un 7% la cantidad de accidentes mortales de motociclistas en el último año, en CABA, respecto al año anterior.**

Definimos a la cantidad de accidentes mortales de motociclistas en siniestros viales como el número absoluto de accidentes fatales en los que estuvieron involucradas víctimas que viajaban en moto en un determinado periodo temporal. Su fórmula para medir la evolución de los accidentes mortales con víctimas en moto es: (Número de accidentes mortales con víctimas en moto en el año anterior - Número de accidentes mortales con víctimas en moto en el año actual) / (Número de accidentes mortales con víctimas en moto en el año anterior) * 100

In [112]:
#Buscamos los datos requeridos para el Kpi que es accidentes mortales en motocolistas

requerimientos= data["Victima"] == "MOTO"
kpi2 =data[requerimientos]


kpi2=  kpi2.groupby("Año")["# Victimas"].sum().reset_index()

#Encontramos la evolucion porcentual de las victimas 
kpi2['Evolucion Porcentual kpi2'] = (((kpi2['# Victimas'].shift(1) - kpi2['# Victimas']) /kpi2['# Victimas'].shift(1)) * 100).fillna(0).round(2)

kpi2

Año  # Victimas  Evolucion Porcentual kpi2
0  2016          67                       0.00
1  2017          62                       7.46
2  2018          61                       1.61
3  2019          50                      18.03
4  2020          30                      40.00
5  2021          46                     -53.33

El KPI propuesto tiene como objetivo reducir en un 7% la cantidad de accidentes mortales de motociclistas en CABA en el último año en comparación con el año anterior. A continuación, se presenta un análisis de los resultados obtenidos:

**Cumplimiento del KPI:**

En los años 2016 y 2017, la evolución porcentual fue del 7.46%, superando el objetivo del 7% de reducción, lo que indica un cumplimiento positivo del KPI en esos años.

**Años 2018 y 2019:**

En 2018, la evolución porcentual fue del 1.61%, lo que sugiere una reducción, pero menor al 7% requerido por el KPI. Sin embargo, en 2019, se observa una evolución porcentual del 18.03%, superando significativamente el objetivo.

**Año 2020:**

En 2020, se destaca una disminución drástica del 40.00%, cumpliendo ampliamente con el KPI y demostrando una reducción sustancial en la cantidad de accidentes mortales de motociclistas.

**Año 2021:**

A pesar de la reducción lograda en 2020, en 2021 se observa una evolución porcentual negativa del -53.33%, indicando un aumento significativo en la cantidad de accidentes mortales. Esta variación negativa no cumple con el KPI y sugiere la necesidad de una revisión de las estrategias implementadas.



Es esencial analizar las tendencias a lo largo de los años para comprender mejor los factores que contribuyen a las variaciones en la evolución porcentual.
La reducción excepcional en 2020 podría estar influenciada por la pandemia de COVID-19 y las restricciones asociadas, lo que llevó a una disminución general en la movilidad.
El aumento abrupto en 2021 destaca la importancia de monitorear continuamente las condiciones del tráfico y de seguridad vial, ajustando las estrategias según sea necesario.
En resumen, mientras que se logró una reducción significativa en 2020, la evolución negativa en 2021 subraya la importancia de mantener un enfoque proactivo en las medidas de seguridad vial para abordar los desafíos cambiantes en la movilidad.

In [113]:
datakpi2 = kpi2.groupby(["Año"])[[ "Evolucion Porcentual kpi2"]].sum().reset_index()
datakpi2

datakpi2.to_csv('datakpi2.csv',index=False, encoding='utf-8')
print(f'Se guardó el archivo ')

Se guardó el archivo 


#  ***Tercer KPI***

**Reducir en un 5% la cantidad de accidentes mortales ocurridos en las avenidas en el último año, en CABA, respecto al año anterior.**

Definimos a la cantidad de accidentes mortales ocurridos en avenidas como el número absoluto de accidentes fatales en las avenidas en un determinado periodo temporal. Su fórmula para medir la evolución de los accidentes mortales en avenidas  es: (Número de accidentes mortales en avenidas en el año anterior - Número de accidentes mortales en avenidas en el año actual) / (Número de accidentes mortales en avenidas en el año anterior) * 100

In [114]:
#Buscamos los datos requeridos para el Kpi que es accidentes mortales en motocolistas

requerimientos= data["Tipo de calle"] == "AVENIDA"
kpi3 =data[requerimientos]


kpi3=  kpi3.groupby("Año")["# Victimas"].sum().reset_index()

#Encontramos la evolucion porcentual de las victimas 
kpi3['Evolucion Porcentual kpi3'] = (((kpi3['# Victimas'].shift(1) - kpi3['# Victimas']) /kpi3['# Victimas'].shift(1)) * 100).fillna(0).round(2)

kpi3

Año  # Victimas  Evolucion Porcentual kpi3
0  2016          93                       0.00
1  2017          93                       0.00
2  2018         101                      -8.60
3  2019          66                      34.65
4  2020          56                      15.15
5  2021          61                      -8.93

La meta de reducir en un 5% la cantidad de accidentes mortales en avenidas en el último año, en la Ciudad Autónoma de Buenos Aires (CABA), en comparación con el año anterior. Los resultados muestran que en los años 2016 y 2017, no hubo cambios en la cantidad de víctimas en avenidas. En 2018, se registró una disminución del -8.60%, superando la meta del KPI. Sin embargo, en 2019, se observó un aumento del 34.65%, contraviniendo la meta establecida. El año 2020 destacó por una disminución del 15.15%, cumpliendo con la meta del KPI. En 2021, se evidenció una disminución del -8.93%, nuevamente superando la meta del KPI. En conjunto, aunque se alcanzaron reducciones notables en 2020 y 2021, es crucial analizar los factores que contribuyeron a las fluctuaciones, especialmente el aumento en 2019, para ajustar estrategias y mantener una mejora sostenida en la seguridad vial en las avenidas de CABA.

In [115]:
datakpi3 = kpi3.groupby(["Año"])[[ "Evolucion Porcentual kpi3"]].sum().reset_index()
datakpi3

datakpi3.to_csv('datakpi3.csv',index=False, encoding='utf-8')
print(f'Se guardó el archivo ')

Se guardó el archivo 
